### PR TITLE
EVEREST-793: metadata service

### PR DESCRIPTION
**Pull request by michal-kralik**
A new endpoint which returns metadata for a  product.

This will be initially used by Everest to determine recommended and supported versions of components during an upgrade.

It's easily extensible to be used by other products as well.

[EVEREST-793](https://perconadev.atlassian.net/browse/EVEREST-793)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | failed |

File: server/metadata_test.go
Comment (recharte) [inline server/metadata_test.go:1]: Maybe we should add one extra test case that covers the case described in the README where the supported version has a semver constraint like `^2.1.0` or `0.0.0 || 1.1.1`.

**Review by recharte — APPROVED** [review]
First time looking at VS source code so I'm not sure I can fully understand the impact of these changes but they look almost completely isolated from the rest of the logic and it looks good :+1:

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

File: sources/metadata/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This directory holds source files for the `metadata` endpoint split by products.
```
Comment (oksana-grishchenko) [inline sources/metadata/README.md:1]: Should we add the real metadata for Everest 0.8.0 in this PR?

**Review by oksana-grishchenko — DISMISSED** [review]
LGTM, great extensible approach👍

**Review by recharte — DISMISSED** [review]

File: sources/metadata/README.md (hunk 1 above)
Comment (michal-kralik) [reply sources/metadata/README.md:1]: It can be added a bit later. There's a process for dev/prod environments.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

**Review by recharte — APPROVED** [review]

**Review by oksana-grishchenko — APPROVED** [review]

**Pull request merged**